File: Pattern40.c
// n=5
// 10101
// 01010
// 10101
// 01010
// 10101

#include<stdio.h>

void main() {
    int n, row, col;
    printf("Enter n: ");
    scanf("%d",&n);
    for(row = 0; row < n; row++) {
        for(col = 0; col < n; col++) {
            if (col%2==0) printf("%d", ((row+1)%2));
            else printf("%d", (row%2));
        }
        printf("\n");
    }
}

File: Pattern40.go
package main

import "fmt"

func main() {
	var n int
	fmt.Print("Enter the row number:")
	fmt.Scanf("%d", &n)

	for i := 1; i <= n; i++ {
		for j := 1; j <= n; j++ {
			if i%2 != 0 {
				if j%2 != 0 {
					fmt.Print(1)
				} else {
					fmt.Print(0)
				}
			} else {
				if j%2 != 0 {
					fmt.Print(0)
				} else {
					fmt.Print(1)
				}
			}
		}
		fmt.Println()
	}
}
